taichi/backends/device.cpp

```diff
@@ -29,12 +31,13 @@ Device::MemcpyCapability Device::check_memcpy_capability(DevicePtr dst,
   }
 
 #if TI_WITH_VULKAN
+#if TI_WITH_LLVM
   if (dynamic_cast<vulkan::VulkanDevice *>(dst.device) &&
       dynamic_cast<cpu::CpuDevice *>(src.device)) {
     // TODO: support direct copy if dst itself supports host write.
     return Device::MemcpyCapability::RequiresStagingBuffer;
   }
-#if TI_WITH_CUDA
+#elif TI_WITH_CUDA
```

Comment:
```suggestion
#if TI_WITH_CUDA
```

Reply:
this is the reason the tests are failing.

Reply:
That makes sense, not sure why I used a #elif, I missed this use case in my testing